src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/EventForDotNet.ts

```diff
@@ -90,40 +214,120 @@ function isCheckbox(element: Element | null) {
 type EventArgsType = 'change' | 'clipboard' | 'drag' | 'error' | 'focus' | 'keyboard' | 'mouse' | 'pointer' | 'progress' | 'touch' | 'unknown' | 'wheel';
 
 export interface UIEventArgs {
-  Type: string;
+  type: string;
 }
 
 interface UIChangeEventArgs extends UIEventArgs {
-  Value: string | boolean;
+  value: string | boolean;
 }
 
 interface UIClipboardEventArgs extends UIEventArgs {
 }
 
 interface UIDragEventArgs extends UIEventArgs {
+  detail: number;
+  dataTransfer: UIDataTransfer;
+  screenX: number;
+  screenY: number;
+  clientX: number;
+  clientY: number;
+  button: number;
+  buttons: number;
+  mozPressure: number;
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
+}
+
+interface UIDataTransfer {
+  dropEffect: string;
+  effectAllowed: string;
+  files: string[];
+  items: UIDataTransferItem[];
+  types: string[];
 }
 
-interface UIErrorEventArgs extends UIEventArgs {
+interface UIDataTransferItem {
+  kind: string;
+  type: string;
+}
+
+interface UIErrorEventArgs extends UIProgressEventArgs {
 }
 
 interface UIFocusEventArgs extends UIEventArgs {
 }
 
 interface UIKeyboardEventArgs extends UIEventArgs {
-  Key: string;
+  char: string;
+  key: string;
+  code: string;
+  location: number;
+  repeat: boolean;
+  locale: string;
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
 }
 
 interface UIMouseEventArgs extends UIEventArgs {
+  detail: number;
+  screenX: number;
+  screenY: number;
+  clientX: number;
+  clientY: number;
+  button: number;
+  buttons: number;
+  mozPressure: number;
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
 }
 
 interface UIPointerEventArgs extends UIMouseEventArgs {
+  pointerId: number;
+  width: number;
+  height: number;
+  pressure: number;
+  tiltX: number;
+  tiltY: number;
+  pointerType: string;
+  isPrimary: boolean;
 }
 
 interface UIProgressEventArgs extends UIEventArgs {
+  lengthComputable: boolean;
+  loaded: number;
+  total: number;
 }
 
 interface UITouchEventArgs extends UIEventArgs {
+  detail: number;
+  touches: UITouchPoint[];
+  targetTouches: UITouchPoint[];
+  changedTouches: UITouchPoint[];
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
+}
+
+interface UITouchPoint {
+  identifier: number;
+  screenX: number;
+  screenY: number;
+  clientX: number;
+  clientY: number;
+  pageX: number;
+  pageY: number;
 }
 
-interface UIWheelEventArgs extends UIEventArgs {
+interface UIWheelEventArgs extends UIMouseEventArgs {
+  deltaX: number;
+  deltaY: number;
+  deltaZ: number;
+  deltaMode: number;
```

Comment:
This all seems like it makes sense to me. /cc @SteveSandersonMS in case there's anything he wants to comment on

Reply:
I made them all camel case as per @SteveSandersonMS request on another issue. They should match the same structure on the c# side.